src/TraceEvent/EventSources/ProcessMetadataEventSource.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Diagnostics.Tracing
+{
+    [EventSource(Name = "ProcessMetadataEventSource")]
+    public sealed class ProcessMetadataEventSource : EventSource
+    {
+        public class Tasks
+        {
+            public const EventTask Process = (EventTask)1;
+            public const EventTask Thread = (EventTask)2;
+            public const EventTask Module = (EventTask)3;
+        }
+
+        [Event(1, Opcode = EventOpcode.Start, Task = Tasks.Process)]
+        public void ProcessStart(long ProcessId, long ParentProcessId, string Executable, string CommandLine)
+        {
+            this.WriteEvent(1, ProcessId, ParentProcessId, Executable, CommandLine);
+        }
+
+        [Event(2, Opcode = EventOpcode.Stop, Task = Tasks.Process)]
+        public void ProcessExit(long ProcessId, long ParentProcessId, int ExitCode, string Executable, string CommandLine)
+        {
+            this.WriteEvent(2, ProcessId, ParentProcessId, ExitCode, Executable, CommandLine);
+        }
+
+        [Event(3, Opcode = EventOpcode.Start, Task = Tasks.Thread)]
+        public void ThreadCreate(long ProcessId, long ThreadId, ulong StackBaseAddress, string ThreadName)
+        {
+            this.WriteEvent(3, ProcessId, ThreadId, StackBaseAddress, ThreadName);
```

Comment:
Do we have a definition of what string to expect in ThreadName? For example .Net has a Thread.Name property and Linux has pthread_getname_np() but I don't know if the strings returned by those APIs are guaranteed to be equal.

Reply:
Are you suggesting a name change? I suppose it could be either, but I think adding two thread name events would be overkill.

Reply:
I am fine with the argument name, I am just trying to sort out what data do we expect to be provided? On Windows we have precedent from the kernel events but on all other platforms we have to define what is the equivalent data we are going to populate this with.